test_end2end/collapse_test/test_common_connection/input.cyp

```diff
@@ -0,0 +1,2 @@
+MERGE (a:P {id: "p1"}) MERGE (b:S {id: "s1"}) CREATE (a)-[e:Edge]->(b);
```

Comment:
I would name this top-level folder `e2e` instead of `test_end2end` because it's shorter and should be obvious what it is.